### ISSUE TITLE
Implement a simple DOM to use with a template

### ISSUE THREAD

**Issue by stadelmanma** [stadelmanma/sablon#17]
Currently the environment instance serves as a poor man's document model by collecting various helper classes like numbering, footnotes and bookmarks. This system is inherently hard to scale. Some work on my live branch has laid a foundation for this such as storing all of the XML files in a template in memory instead of simple sequential processing.

The general implementation would be as followd:
 1. Add a`lib/sablon/document_object_model` directory
 2. In that directory class files could be written to handle specific aspects of the document, such as numbering.xml. A generic "document" class would also exist.
3. The files that get shuttled off to the various DOM subclasses would be defined in a single "initialization" method of some sort.
4. A general DOM class instance would also exist that does the '"thinking" and would contain methods such as `add_image`, `add_relationship`, `add_bookmark`, `add_list_definition`, etc. 
    * The implementation details of these methods may be split between the "DOM" class and the file specific sub class. Adding relationships would be an example since the DOM class would need to pick the proper file (i.e. document.xml => document.xml.rels) due to the fact a file specific subclass would have no knowledge of the source of the relationship to be added.

Examples of files that might be repurposed into "dom classes" are lib/sablon/numbering.rb and lib/sablon/relationships.rb

**Comment by stadelmanma** [comment]
Given the flexibility this would add with relationships I may need to create a missing file, below is an XML sample that should work. I don't know where best to store this type of information but it's an easy case to test for by adding something like a link to a header or footer.

```XML
<?xml version="1.0" encoding="UTF-8"?>
<Relationships xmlns="http://schemas.openxmlformats.org/package/2006/relationships">
</Relationships>
```

**Comment by stadelmanma** [comment]
Fixed in upstream via #85
